engine/src/test/java/io/zeebe/engine/processor/StreamProcessorReprocessingTest.java

```diff
@@ -395,8 +390,10 @@ public void processRecord(
     streamProcessorRule.writeWorkflowInstanceEvent(ELEMENT_ACTIVATING, 1);
     final long snapshotPosition =
         streamProcessorRule.writeWorkflowInstanceEvent(ELEMENT_ACTIVATING, 1);
+    streamProcessorRule.writeWorkflowInstanceEvent(ELEMENT_ACTIVATING, 1);
+    // we write three event to make sure that the second event was committed and is part of the
+    // state (lastProcessedEvent)
     processingLatch.await();
-    streamProcessorRule.waitForNextSnapshot();
```

Comment:
Do we have another test that verifies processor does not reprocess event at the last processed position in snapshot?

Reply:
I thought but I can check

Reply:
Ok I remember I rewrote the test that such that the event is part of the state/db because this is the only thing what matters. Here we verify that we dont reprocess again the events, which we already saw and part of the db. This is unrelated of how and when we take snapshot since this is no longer part of the stream processor. It only matters what is part of the db.

Reply:
How are we ensuring that the event is part of the state and the reprocessor is reading the position from the state instead of reading the log from the beginning?

Reply:
> How are we ensuring that the event is part of the state and the reprocessor is reading the position from the state instead of reading the log from the beginning?

We write three events. Await processing of the third, which means second needs to be committed and written to the DB. On restart we check the processed events and check that the second, which should be part of the db is not part of the processed list.